lib/enhancer.py

```diff
@@ -1,80 +1,68 @@
-import json
-from lxml import etree
-from itertools import repeat
+import os
 
 from helpers.errorHelpers import OCLCError
 from helpers.logHelpers import createLog
-from lib.dataModel import Agent
+from lib.dataModel import Agent, Identifier
 from lib.readers.oclcClassify import classifyRecord
 from lib.parsers.parseOCLC import readFromClassify
 from lib.kinesisWrite import KinesisOutput
 
 logger = createLog('enhancer')
 
 def enhanceRecord(record):
+    """Takes a single input record and retrieves data from the OCLC Classify
+    service. Manages the overall workflow of the function."""
+
+    try:
+        sourceData = record['data']
+    except KeyError:
+        logger.error('Missing data from input event')
+        return False
+
     try:
-        source = record['source']
-        recID = record['recordID']
-        data = record['data']
+        workUUID = sourceData['uuid']
+        searchType = sourceData['type']
+        searchFields = sourceData['fields']
     except KeyError as e:
         logger.error('Missing attribute in data block!')
         logger.debug(e)
         return False
+    except TypeError as e:
+        logger.error('Could not read data from source')
+        logger.debug(e)
+        return False
 
-    logger.info('Starting to enhance record {} from {}'.format(recID, source))
-    try:
+    logger.info('Starting to enhance work record {}'.format(workUUID))
 
+    try:
         # Step 1: Generate a set of XML records retrieved from Classify
         # This step also adds the oclc identifiers to the sourceData record
-        classifyData, sourceData = classifyRecord(source, data, recID)
-
-        if classifyData is not None:
-            # Step 2: Parse the data recieved from Classify into the SFR data model
-            parsedData = readFromClassify(classifyData)
-
-            # Step 3: Merge this data with the source data
-            mergedData = mergeData(sourceData, parsedData)
-        else:
-            mergedData = sourceData
-
-        # Step 4: Output this block to kinesis
-        KinesisOutput.putRecord(mergedData)
+        classifyData = classifyRecord(searchType, searchFields, workUUID)
+
+        if classifyData is None:
```

Comment:
Probably beyond the scope is this PR, but this feels like a potentially confusion side-effect, as does the `putRecord()` As I understand the return values `False` means there was some kind of error, `True` means one kind of record was stored, and `None` means either nothing happened  or multiple records were queued for later? Maybe this needs to be broken down another way?

Reply:
Agreed. I'm going to refactor this so instead of returning `False` or `None` it raises an exception in those cases, which is caught and logged in a similar fashion to what you suggested above